ls1intum/Artemis
# Add filter for "Build Failed" to exercise scores for programming exercises

### Checklist

### Motivation and Context
This PR solves the issue #1010 by providing a filter for participations with failed submission. It also adds another filter option to the scores page to filter for submissions with failed build.

### Description
1. Add radio button "Show Build Failed" to filter bar on exercises page (if it is a programming 
  exercise)
1.1 Make sure responsive display of filters and buttons looks good (only show them in the same line if window width allows it, consider that width needs to be much wider for German site)
2. Add radio buttons "Show all" and "Show Failed" to participations page for programming exercises

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Administration
3. Click on a course that has a programming exercise
4. Click on "Scores" for a programming exercise
4.1. Click on radio button "Show Build Failed" (under the headline)
4.2. Verify that only results with failed builds are shown
5. Click on "Participations" for a programming exercise
5.1. Click on radio button "Show Failed" (under the headline)
5.2. Verify that only results with failed submissions are shown

### Screenshots
New filter option "Show Build Failed" (only shown for scores for programming exercises):
![OnPaste 20191215-174058](https://user-images.githubusercontent.com/7109225/70865816-2074c100-1f62-11ea-9374-52c995a11a12.png)

Two rows for filters and buttons if window width is too small:
![OnPaste 20191215-174120](https://user-images.githubusercontent.com/7109225/70865818-21a5ee00-1f62-11ea-81b5-221cb3883923.png)

New filter option "Show Failed" (only shown for participations for programming exercises):
![OnPaste 20191216-000336](https://user-images.githubusercontent.com/7109225/70870758-84fe4300-1f97-11ea-9531-a6f08bf9a180.png)
